blivet/blivet.py

```diff
@@ -786,12 +786,22 @@ def create_device(self, device):
             :type device: :class:`~.devices.StorageDevice`
             :rtype: None
         """
-        self.devicetree.actions.add(ActionCreateDevice(device))
+        action_create_dev = ActionCreateDevice(device)
+        self.devicetree.actions.add(action_create_dev)
 
         is_snapshot = isinstance(device, LVMLogicalVolumeDevice) and device.is_snapshot_lv
 
         if device.format.type and not device.format_immutable and not is_snapshot:
-            self.devicetree.actions.add(ActionCreateFormat(device))
+            action_create_fmt = None
+            try:
+                action_create_fmt = ActionCreateFormat(device)
+            except (ValueError, DependencyError) as e:
+                # revert devicetree changes done so far
+                if action_create_fmt is not None:
+                    self.devicetree.actions.remove(action_create_fmt)
```

Comment:
This `remove` call isn't necessary and will trigger an exception, I believe, since `action_create_fmt` has not been added at this point.

Reply:
Removing the `ActionCreateFormat` won't remove the device from the tree. (Removing the `ActionCreateDevice`, later, will.) You didn't add `action_create_fmt` yet, so there is no need to remove it.

Reply:
Ah. Now I see. You are right. My bad. Fixed.